crates/ruff_cli/src/cache.rs

```diff
@@ -233,7 +233,7 @@ pub fn get(
                 messages.push(Message {
                     kind: header.kind,
                     range: header.range,
-                    fix: header.fix,
+                    fix: Some(header.fix),
```

Comment:
`MessageHeader` is a helper struct to deserialize a `Message`. Its fields must match the fields on `Message` or the deserialization will fail if we have a `Message` without a header. 

That's why we should change the `fix` definition on `MessageHeader` to `Option<Fix>` to match `Message`

```rust 
#[derive(Deserialize)]
struct MessageHeader {
    kind: DiagnosticKind,
    range: TextRange,
    fix: Option<Fix>,
    file_id: usize,
    noqa_row: TextSize,
}
```

Reply:
Do you normally add tests for things like this to ensure it stays up to date?

Reply:
Thx. We probably should but don't have to as part of this PR.